master/internal/api_project.go

```diff
@@ -37,6 +38,69 @@ func (a *apiServer) GetProjectByID(
 	return p, nil
 }
 
+func (a *apiServer) getProjectColumnsByID(
+	ctx context.Context, id int32, curUser model.User,
+) (*apiv1.GetProjectColumnsResponse, error) {
+	notFoundErr := status.Errorf(codes.NotFound, "project (%d) not found", id)
+	p := &projectv1.Project{}
+	if err := a.m.db.QueryProto("get_project", p, id); errors.Is(err, db.ErrNotFound) {
+		return nil, notFoundErr
+	} else if err != nil {
+		return nil, errors.Wrapf(err, "error fetching project (%d) from database", id)
+	}
+	if ok, err := project.AuthZProvider.Get().CanGetProject(ctx, curUser, p); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, notFoundErr
+	}
+	// Get general columns
+	generalColumns := make([]projectv1.GeneralColumn, 0, len(projectv1.GeneralColumn_value))
+	for gc := range projectv1.GeneralColumn_value {
+		generalColumns = append(
+			generalColumns, projectv1.GeneralColumn(projectv1.GeneralColumn_value[gc]))
+	}
+	// Get hyperpatameters columns
+	hyperparameters := struct {
+		Hyperparameters []string
+	}{}
+	err := db.Bun().
+		NewSelect().Table("projects").Column("hyperparameters").Where(
+		"id = ?", id).Scan(ctx, &hyperparameters)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get metrics columns
+	metricNames := []struct {
+		Tname []string
+		Vname []string
+	}{}
+	metricColumns := make([]string, 0)
+	err = db.Bun().
+		NewSelect().
+		TableExpr("exp_metrics_name").
+		TableExpr("LATERAL json_array_elements_text(vname) AS vnames").
+		TableExpr("LATERAL json_array_elements_text(tname) AS tnames").
+		ColumnExpr("array_to_json(array_agg(DISTINCT tnames)) AS tname").
+		ColumnExpr("array_to_json(array_agg(DISTINCT vnames)) AS vname").
+		Where("project_id = ?", id).Scan(ctx, &metricNames)
+	if err != nil {
+		return nil, err
+	}
+	for _, mn := range metricNames {
+		for _, mnt := range mn.Tname {
+			metricColumns = append(metricColumns, fmt.Sprintf("%s.%s", "training", mnt))
```

Comment:
question: what do we get out of indexing the metric names and is `.` a standard notation for us? is it going to be parsed? 

```
  "metrics": [
    "training.loss1",
    "training.loss2",
    "validation.loss1",
    "validation.loss2"
  ]
  ```